### PR TITLE
fix ssh into windows

### PR DESCRIPTION
**Pull request by ricardosantos9521**
- Fix regex expression that was trying to detect if uri.path was a windows path.
- use resolveSSHHostName to detect username if needed in windows remotes too
- convert uri.path to a windows path

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -57,12 +57,13 @@ async function openWindowsTerminal(profile: IWTProfile, uri?: vscode.Uri) {
...
       const host = uri.authority.split('+')[1];
-      const isWindows = !!uri.path.match(/^\/[a-z]:\//);
+      const isWindows = !!uri.path.match(/^\/[a-zA-Z]:\/.*$/);
+      const remoteMachine = await resolveSSHHostName(host);
       if (isWindows) {
-        // Changing paths on Windows seems tricky
-        args.push('ssh', host);
+        // remove first /
+        const uri_windows = uri.path.substring(1, uri.path.length);
+        args.push('ssh', remoteMachine, `cd ${uri_windows} && powershell.exe`);
```
Comment (Tyriar) [inline src/extension.ts:65]: I don't think this is reliable enough unfortunately as `&&` may not work in the shell. sshing in will either run in cmd or powershell and from my experiments I couldn't figure out how to link 2 commands together in cmd with ssh liking it. Something I have yet to try is wrapping it in a `powershell -Command` to re-launch the shell which might work.

File: src/extension.ts [hunk 2]
```diff
@@ -57,12 +57,13 @@ async function openWindowsTerminal(profile: IWTProfile, uri?: vscode.Uri) {
     if (uri.authority.startsWith('ssh-remote+')) {
       // Handle SSH remotes first
       const host = uri.authority.split('+')[1];
-      const isWindows = !!uri.path.match(/^\/[a-z]:\//);
+      const isWindows = !!uri.path.match(/^\/[a-zA-Z]:\/.*$/);
+      const remoteMachine = await resolveSSHHostName(host);
```
Comment (Tyriar) [inline src/extension.ts:61]: 👍 can you split this out into a separate PR? The uri thing might take a bit to get right.

File: src/extension.ts (hunk 1 above)
Comment (ricardosantos9521) [reply src/extension.ts:65]: i think we should probably use cmd.exe /c "cd ... && shell" wt.exe creates multiples panels when sees a ; even inside a string. i will investigate this.
